tools/backups2datalad/datasetter.py

```diff
@@ -399,21 +399,17 @@ def commit_has_assets(commit_hash: str) -> bool:
     @cached_property
     def gh(self) -> Github:
         token = readcmd("git", "config", "hub.oauthtoken")
-        return Github(token)
+        return Github(
+            token,
+            retry=Retry(
+                total=12,
+                backoff_factor=1.25,
+                status_forcelist=[500, 502, 503, 504],
+            ),
+        )
 
     def get_github_repo(self, repo_fullname: str) -> Repository:
-        i = 0
-        while True:
-            try:
-                return self.gh.get_repo(repo_fullname)
-            except ConnectionError:
-                log.warning(
-                    "Request to fetch %s GitHub repository details resulted in"
-                    " connection error; waiting & retrying",
-                    repo_fullname,
-                )
-                i += 1
-                sleep(i * i)
+        return self.gh.get_repo(repo_fullname)
```

Comment:
what about a generic ConnectionError -- wouldn't it just bubble up now?

Reply:
The addition of the `Retry` above will cause ConnectionErrors and various other errors to be retried until we run out of retries.